packages/openapi-v3/src/controller-spec.ts

```diff
@@ -77,6 +77,25 @@ function resolveControllerSpec(constructor: Function): ControllerSpec {
     spec = {paths: {}};
   }
 
+  const isClassDeprecated = MetadataInspector.getClassMetadata<boolean>(
+    OAI3Keys.DEPRECATED_CLASS_KEY,
+    constructor,
+  );
+
+  if (isClassDeprecated) {
+    debug('  using class-level @deprecated()');
+  }
+
+  if (isClassDeprecated) {
+    for (const path of Object.keys(spec.paths)) {
+      for (const method of Object.keys(spec.paths[path])) {
+        if (isClassDeprecated) {
```

Comment:
I think the 2nd `isClassDeprecated` check is a duplicate.

Reply:
Changed!  The top if block is for `if (isClassDeprecated || classHasTags) {}`.  Preferring to iterate over existing paths once, and not have two `for (path of spec.paths) {}` loops.